components/portal/messages/controllers.js

```diff
@@ -94,33 +103,90 @@ define(['angular'], function(angular) {
             .catch(filterMessagesFailure);
           };
 
-
         /**
-         * Separate the message types in scope for child controllers
-         * @param {Object} result
+         * Filters raw messages by group membership and
+         * valid begin/end dates.
+         * Resolves URLs in messages.
+         *
+         * @param {Object} messages
          */
-        var filterMessagesSuccess = function(result) {
-          $scope.seenMessageIds = result[0];
-          var dataMessages = result[1];
-          var groupedMessages = result[2];
-          var filteredMessages = $filter('filterForCommonElements')(
-            groupedMessages,
-            dataMessages
-          );
-          var dateFilteredMessages =
-            $filter('filterByDate')(filteredMessages);
-            $scope.messages =
-              $filter('separateMessageTypes')(dateFilteredMessages);
-          $scope.hasMessages = true;
+        var configureMessages = function(messages) {
```

Comment:
Is `configure` the best verb to characterize what's happening here? Granting it's more than `filter` since it resolves URLs in messages. "Resolve"? "Inflate"? "Personalize"?